### PR TITLE
Adding ZeroBrane Studio

### PR DESCRIPTION
**Pull request by MrSyabro**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: zbstudio.appdata.xml.patch [hunk 1]
```diff
@@ -0,0 +1,4 @@
+3c3
```
Comment (hfiguiere) [inline zbstudio.appdata.xml.patch:1]: All patches should be created with `-u`

File: com.zerobrane.studio.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+      "name": "zbstudio",
+      "buildsystem": "simple",
+      "build-commands": [
+        "pwd",
+        "sh ZeroBraneStudioEduPack-1.90-linux.sh --keep --noexec --nox11",
+        "patch -t ZeroBraneStudio-linux/install.sh install.sh.patch",
+        "patch -t ZeroBraneStudio-linux/app/zbstudio.sh zbstudio.sh.patch",
+        "patch -t ZeroBraneStudio-linux/app/zbstudio/res/zbstudio.appdata.xml zbstudio.appdata.xml.patch",
+        "patch -t ZeroBraneStudio-linux/app/zbstudio/res/zbstudio.desktop zbstudio.desktop.patch",
```
Comment (hfiguiere) [inline com.zerobrane.studio.json:25]: you should just use `desktop-file-edit` here.

**Comment by hfiguiere** [comment]
why not building from source instead?

**Comment by MrSyabro** [comment]
> why not building from source instead?

Good question.  I'll fix everything.

File: com.zerobrane.studio.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/* Automatically generated from com.github.MrSyabro.zbstudio.yml, do not modify */
```
Comment (barthalion) [inline com.zerobrane.studio.json:1]: FYI we also accept YAML manifests.

**Comment by MrSyabro** [comment]
> why not building from source instead?

The source code repository contains many binaries compiled for different platforms.

File: com.zerobrane.studio.json (hunk 3 above)
Comment (MrSyabro) [reply com.zerobrane.studio.json]: [e0bd1b8](https://github.com/flathub/flathub/pull/3877/commits/e0bd1b849341378f96f0ef92061ebd02e026f1b5)

File: zbstudio.appdata.xml.patch (hunk 1 above)
Comment (MrSyabro) [reply zbstudio.appdata.xml.patch]: ok

File: com.zerobrane.studio.json (hunk 2 above)
Comment (MrSyabro) [reply com.zerobrane.studio.json]: Ok

File: com.zerobrane.studio.json (hunk 3 above)
Comment (MrSyabro) [reply com.zerobrane.studio.json]: Я перевел в yaml

Comment (MrSyabro) [reply com.zerobrane.studio.json]: I translated to yaml

**Comment by MrSyabro** [comment]
bot, build com.zerobrane.studio

**Comment by barthalion** [comment]
```
builddir/files/share/appdata/com.zerobrane.studio.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
```

**Comment by farribeiro** [comment]
Extra-data?

**Comment by MrSyabro** [comment]
> Extra-data?

What do you have in mind?

**Comment by farribeiro** [comment]
> > Extra-data?
> 
> What do you have in mind?

I suspected that the app should be installed via extra-data , that is if it was proprietary

**Comment by MrSyabro** [comment]
> I suspected that the app should be installed via extra-data , that is if it was proprietary

Installation is done using the official self-extracting sh script. This is how this open source software is distributed

**Comment by barthalion** [comment]
Any reason you have not contacted upstream about your submission?

The app itself seems easy to build from source: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zerobrane-studio. It would be preferred to do that if it doesn't try to access network during build.

**Comment by MrSyabro** [comment]
> Any reason you have not contacted upstream about your submission?
> 
> The app itself seems easy to build from source: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zerobrane-studio. It would be preferred to do that if it doesn't try to access network during build.

This software has no source code to build. In the project repository, the same as in the archive from off. site + executable files for other platforms.

**Comment by hfiguiere** [comment]
So project license `MIT` is a lie?

**Comment by MrSyabro** [comment]
> So project license `MIT` is a lie?

No, why not?

**Comment by MrSyabro** [comment]
The installation method through the archive from the official site is no different from the installation method from the repository. The source code for this project does not require compilation.

**Comment by barthalion** [comment]
I'm surprised but it seems to be the case at first glance. That being said, it is expected from submitters to contact project developers nonetheless.

**Comment by MrSyabro** [comment]
> I'm surprised but it seems to be the case at first glance. That being said, it is expected from submitters to contact project developers nonetheless.

What should I tell the developers about?

**Comment by barthalion** [comment]
Ask whether they're interested in officially maintaining their app on Flathub.

**Pull request closed without merging**